src/components/ReorderableList.jsx

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { Button, Space } from 'antd';
+import { UpOutlined, DownOutlined } from '@ant-design/icons';
+
+const moveUp = (source, id) => {
+  const index = source.findIndex((e) => e.key === id);
+
+  const arr = [...source];
+
+  if (index <= 0) {
+    return arr;
+  }
+
+  const el = arr[index];
+  arr[index] = arr[index - 1];
+  arr[index - 1] = el;
+
+  return arr;
+};
+
+const moveDown = (source, id) => {
+  const index = source.findIndex((e) => e.key === id);
+
+  const arr = [...source];
+
+  if (index === -1 || index >= source.length - 1) {
+    return arr;
+  }
+
+  const el = arr[index];
+  arr[index] = arr[index + 1];
+  arr[index + 1] = el;
+
+  return arr;
+};
+
+const ReorderableList = (props) => {
+  const {
+    onMoveUp, onMoveDown, reorderableList, leftItem, rightItem,
+  } = props;
+
+  const upButton = (key, currentPosition) => (
+    <Button
+      size='small'
+      shape='circle'
+      disabled={currentPosition === 0}
+      icon={<UpOutlined />}
+      style={{ marginLeft: '5px' }}
+      onClick={() => {
+        onMoveUp(moveUp(reorderableList, key));
+      }}
+    />
+  );
+
+  const downButton = (key, currentPosition) => (
+    <Button
+      size='small'
+      shape='circle'
+      disabled={currentPosition === reorderableList.length - 1}
+      icon={<DownOutlined />}
+      style={{ marginRight: '5px' }}
+      onClick={() => {
+        onMoveDown(moveDown(reorderableList, key));
+      }}
+    />
+  );
+
+  const fullItem = (listItem, i) => (
```

Comment:
I think we can do better in terms of descriptive naming in here. 
In terms of `leftItem` and `rightItem`: I took a look at the code, `rightItem` seems to be the actual item name, so I would just call it `itemName` instead of `rightItem`. `leftItem` is the switch that appears in metadata settings, but not in group by settings. It is hard to name that one, but I would maybe call it `itemProps`? This way the name signifies that it can be not only a `switch` but something else, and that whatever it is, belongs to the item. Happy to accept other naming suggestions.
In terms of the function `fullItem`, since its functionality is to bring the whole item together with all its required buttons, name, etc, I would probably call it `composeItem`.

Reply:
I changed `listItem` to `itemData` and `fullItem` to `composeItem`. 

About `leftItem` and `rightItem`, they can be any kind of component interchangeably, the only difference between them is that one is to the left of the up down buttons and the other one is to the right, that is why I feel that naming fits them better.

This way we can keep it more abstract so that whoever is using it can choose whether it is going to have a name or something more complex and in any side they prefer without really having a downside on its current usage (I believe).

Reply:
Sounds good Martin, let's keep it in the way you are proposing, what you are saying makes sense.